jhub_apps/service/routes.py

```diff
@@ -160,27 +160,67 @@ async def create_server(
 @router.post("/server/")
 @router.post("/server/{server_name}")
 async def start_server(
-    server_name=None,
+    server_name: str = None,
     user: User = Depends(get_current_user),
 ):
     """Start an already existing server."""
-    logger.info("Starting server", server_name=server_name, user=user.name)
+    logger.info(f"Starting server with name '{server_name}' for user '{user.name}'")
+    
     hub_client = HubClient(username=user.name)
+
+    # Check if the server is shared and whether the user has permission to start it
+    try:
+        shared_servers = get_shared_servers(current_hub_user=user)
+        if server_name and any(server['name'] == server_name for server in shared_servers):
+            # User is trying to start a shared server without permission
+            raise HTTPException(
+                detail=f"User '{user.name}' does not have permission to start server '{server_name}'",
+                status_code=status.HTTP_403_FORBIDDEN,
+            )
+    except ValueError as e:
+        logger.error(f"Error in shared servers check: {e}")
+        raise HTTPException(
+            detail=f"Failed to check shared servers: {e}",
+            status_code=status.HTTP_400_BAD_REQUEST,
+        )
+    
```

Comment:
This permissions part needs to be handled on JupyterHub side (as JupyterHub knows what permissions a user have and jhub-apps service doesn't), jhub-apps service is not responsible for this and besides the checks done here is not sufficient to determine if the user has permissions to start the server or not. We actually just need to call `hub_client.start_server` and based on the response to it, we need to return appropriate response.

Reply:
Updated.